### ISSUE TITLE
Attributes update

### ISSUE THREAD

**Issue by shawncplus** [shawncplus/ranviermud#105]
All attributes should follow a pattern similar to below.

damage/healing scenario

```
baseHealth: 100
maxHealth: 100
healthDelta: Math.min(0, healthDelta)
health = Math.max(maxHealth, maxHealth + healthDelta) = 100

player takes 25 damage

baseHealth = 100
maxHealth = 100
healthDelta = Math.min(0, -25) = -25
health = health = Math.max(maxHealth, maxHealth + healthDelta) = 75

player heals for 100

baseHealth = 100
maxHealth = 100
healthDelta = Math.min(0, 75) = 0
health = health = Math.max(maxHealth, maxHealth + healthDelta) = 100
```

temporary stat debuff scenario (-5 str for 5 seconds, for example)

In practice pools (health/mana/energy) should be the only stats where the delta changes. Attributes (str, int, wis, etc.) buffs/debuffs should modify only the max property
```
baseStr = 10
maxStr = 5
delta = 0    // stat buffs/debuffs modify max _not_ delta
str = 5
```

`delta` is essentially a effect-agnostic modification to a stat. So there would be no reason for an effect to ever directly modify delta. Effects _could_ cause damage but that would happen by them firing damage events which _in turn_ modify delta

**Comment by marado** [comment]
> Math.max(0, -25) = -25

You're doing it wrong! ;-)

**Comment by shawncplus** [comment]
https://www.youtube.com/watch?v=PibDMGxiyJw

**Comment by shawncplus** [comment]
Closed with #107
